master/internal/api_trials.go

```diff
@@ -877,15 +877,15 @@ func (a *apiServer) GetValidationMetrics(
 		return resp.Send(&apiv1.GetValidationMetricsResponse{Metrics: m})
 	}
 	if err := a.streamMetrics(resp.Context(), req.TrialIds, sendFunc,
-		model.ValidationMetricType.ToString()); err != nil {
+		model.ValidationMetricType); err != nil {
 		return err
 	}
 
 	return nil
 }
 
 func (a *apiServer) streamMetrics(ctx context.Context,
-	trialIDs []int32, sendFunc func(m []*trialv1.MetricsReport) error, metricType string,
+	trialIDs []int32, sendFunc func(m []*trialv1.MetricsReport) error, metricType model.MetricType,
```

Comment:
how are we going to return generic metrics with arbitrary user-specified type `inference`?

Reply:
would that just be accepting those arbitrary strings as MetricType? I'm not sure if I'm getting your comment correctly

Reply:
yes, I believe metric type has to be an arbitrary string. many of these APIs have `metricType string` today specifically because we were expecting we'll have arbitrary types.

Reply:
we chatted offline. the existing `MetricType` works here.